snuba/writer.py

```diff
@@ -58,8 +60,41 @@ def __encode(self, row):
 
     def write(self, rows):
         parameters = self.__options.copy()
-        parameters['query'] = "INSERT INTO {table} FORMAT JSONEachRow".format(table=self.__schema.get_table_name())
+        parameters['query'] = "INSERT INTO {table} FORMAT JSONEachRow".format(table=self.__table_name)
         requests.post(
             urljoin(self.__base_url, '?' + urlencode(parameters)),
             data=map(self.__encode, rows),
         ).raise_for_status()
+
+
+class BufferedWriterWrapper:
+    """
+    This is a wrapper that adds a buffer around a BatchWriter.
+    When consuming data from Kafka, the buffering logic is performed by the
+    batching consumer.
+    This is for the use cases that are not Kafka related.
+
+    This is not thread safe. Don't try to do parallel flush hoping in the GIL.
+    """
+
+    def __init__(self, writer: BatchWriter, buffer_size: int):
+        self.__writer = writer
+        self.__buffer_size = buffer_size
+        self.__buffer: List[Mapping[str, Any]] = []
```

Comment:
It might make sense to type alias this as `Row` and then use it in all the `Writer` implementations for consistency.

Reply:
ok